packages/elements/src/slider/index.ts

```diff
@@ -575,29 +581,99 @@ export class Slider extends ControlElement {
   }
 
   /**
-   * Calculate percentage from value, min and max
-   * @param value value to be calculated
-   * @returns percentage
+   * Handles key down event on thumbs
+   * @param event Keyboard event
+   * @returns {void}
    */
-  private calculatePercentage (value: number): number {
-    const percentage = Math.abs((((value || 0) - this.minNumber) / (this.maxNumber - this.minNumber)) * 100);
-    if (percentage > 100) {
-      return 100;
+  private onKeyDown (event: KeyboardEvent): void {
+    if (this.readonly || this.minNumber === this.maxNumber) {
+      return;
+    }
+
+    switch (event.key) {
+      case 'ArrowDown':
+      case 'Down':
+      case 'ArrowLeft':
+      case 'Left':
+        this.onApplyStep(Direction.Down);
+        break;
+      case 'ArrowUp':
+      case 'Up':
+      case 'ArrowRight':
+      case 'Right':
+        this.onApplyStep(Direction.Up);
+        break;
+      default:
+        return;
+    }
+
+    event.preventDefault();
+  }
+
+  /**
+   * Increase or decrease value depending on direction
+   * And fires value change event
+   * @param direction Up or Down
+   * @returns {void}
+   */
+  private onApplyStep (direction: Direction): void {
+    const currentPercentageValue = this.calculatePercentage(this.valueNumber, 1);
+    const percentageStep = this.calculatePercentage(this.minNumber + this.stepRange, 1);
+
+    const percentageValue = direction === Direction.Up ? currentPercentageValue + percentageStep : currentPercentageValue - percentageStep;
+    const nearestPossibleValue = this.getNearestPossibleValue(percentageValue);
+
+    const value = this.getValueFromPercentage(nearestPossibleValue);
+
+    this.value = this.format(value);
```

Comment:
It is called `dispatchDataChangedEvent` because it supports data like `value`, `from` and `to` in one go and while it is not very elegant it checks for data change inside the function.

Reply:
Confusing, but OK :)